xla/service/collective_quantizer_test.cc

```diff
@@ -233,6 +233,30 @@ TEST_F(CollectiveQuantizerTest, AllGatherQuantizeUnary) {
 }
 
 TEST_F(CollectiveQuantizerTest, AllGatherQuantizeMultiUser) {
```

Comment:
Why explicitly have both a replicated and non-replicated test when both don't change the HLO for the same reason, which is because the all-gather has multiple users and not because the scale is non-replicated?

Instead, I'd have a test that normally would be rewritten, except for the fact the scale is non-replicated.